src/Features/LanguageServer/Protocol/RequestDispatcher.cs

```diff
@@ -105,7 +140,7 @@ public TestAccessor(RequestDispatcher requestDispatcher)
                 => _requestDispatcher = requestDispatcher;
 
             public IRequestHandler<RequestType, ResponseType> GetHandler<RequestType, ResponseType>(string methodName)
-                => (IRequestHandler<RequestType, ResponseType>)_requestDispatcher._requestHandlers[methodName].Value;
+                => (IRequestHandler<RequestType, ResponseType>)_requestDispatcher._requestHandlers.Single(handler => handler.Key.MethodName == methodName).Value.Value;
```

Comment:
Just making sure, didn't the comment say this system could support multiple handlers for the same method, making the `Single` here bad? or is that not true per dispatcher, and only across the whole system?

Reply:
Multiple handlers for the same method across the whole system.  Each dispatcher (aka LSP server) can only have a single handler for a method.